rfc/0047-unified-user-agent.md

```diff
@@ -0,0 +1,166 @@
+# Meta
+
+* RFC Name: unified-user-agent
+* RFC ID: 47
+* Start Date: 2019-02-11
+* Owner: Sergey Avseyev
+* Contributors: Michael Nitschinger, Sergey Avseyev
+* Current Status: Draft
+
+# Summary
+
+This RFC proposes the unification across SDKs of the “User Agent” identifier
+which they send to the server on numerous occasions.
+
+# Motivation
+
+At the time of writing there are two ways the SDKs send identifying information
+to the server: when sending a request via HTTP to services and during KV
+"Hello" negotiation. While both are slightly different on the wire, they serve
+a common purpose:
+
+Identifying and "fingerprinting" the client to the server to aid subsequent
+debugging and after-the-fact information gathering (i.e. "cbcollectinfo").
+
+As a result, the more unified the user agent is across SDKs, the easier it is
+for the field and automated tooling to extract and parse this information.
+
+# General Design
+
+There is already a very prominent precedent for user agents: The HTTP
+User-Agent header (defined here
+[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent)).
+
+The SDKs send an actual HTTP User-Agent header to services like query or
+analytics, but they also need to send a trimmed-down version via a KeyValue
+HELLO negotiation. As a result, this RFC defines a more terse version of the
+common HTTP equivalent:
+
+    <sdk-identifier>/<version> [(<system-information>)]
+
+The following sections break down each individual piece.
+
+## SDK-Identifier
+
+* Required
+
+The identifier allows to uniquely distinguish between the different SDKs used.
+The following identifiers must be followed by the implementations:
+
+| SDK           | Identifier |
+| ------------- | ---------- |
+| .NET          | `dotnet`   |
+| C++           | `cxx`      |
+| Go            | `golang`   |
+| JVM Core      | `jvm-core` |
+| Java          | `java`     |
+| Kotlin        | `kotlin`   |
+| Libcouchbase  | `lcb`      |
+| Node.js       | `nodejs`   |
+| PHP           | `php`      |
+| Python        | `python`   |
+| Ruby          | `ruby`     |
+| Rust          | `rust`     |
+| Scala         | `scala`    |
+| Kafka         | `kafka`    |
+| ElasticSearch | `es`       |
+
```

Comment:
```suggestion
| Java Columnar      | `couchbase-java-columnar` |
| Node.js Columnar   | `nodejs-columnar`         |
| Python Columnar    | `python-columnar`         |

```

Seems like we might as well add the Columnar SDKs since we are making an update when they are already in existence.